src/components/FullDetailView/InnerDataSection.js

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation and others. Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+import React from 'react'
+import PlaceholderRenderer from '../Renderers/PlaceholderRenderer'
+import MonacoEditorWrapper from '../MonacoEditorWrapper'
+
+const InnerDataSection = ({ value, name, type = 'json', actionButton = null }) => {
```

Comment:
What i've done here is to move all the render functions that previously were into the FullDetailComponent.
Since the component was more than 380 lines of codes, i've moved into separate presentational components all the render functions, in order to have the code more clean.

All the components that are used specifically by FullDetailsView are placed in the same folder, other components that are reusable everywhere (like Labels, ButtonWithTooltip) are in a more generic folder

Reply:
+1 on the refactoring. Great idea.

I disagree however with the grouping. These components just **happen** to currently only be used in the full details view. It is quite reasonable to use them elsewhere. Assuming there is no significant simplification or economy we get from positioning them as "part of" the full details view, what do we lose from positioning them more generally? IMO it is completely legit to imagine the HeaderSection or LicenseSection (for example) being used in some other context where we want to show details.

Reply:
It makes sense, but in this case a different naming should be applied to those components.
`HeaderSection` could have a meaning if positioned under the FullDetailsView folder, but it hasn't the same meaning when used a reusable component.